src/api.rs

```diff
@@ -2432,6 +2466,39 @@ pub struct Monitor {
     pub status: String,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum ApiMonitorCheckInStatus {
+    Unknown,
+    Ok,
+    InProgress,
+    Error,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct ApiMonitorCheckIn {
+    pub id: Uuid,
+    pub status: Option<ApiMonitorCheckInStatus>,
+    pub duration: Option<u64>,
+    pub environment: Option<String>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct ApiCreateMonitorCheckIn {
+    pub status: ApiMonitorCheckInStatus,
+    pub environment: String,
+}
+
+#[derive(Debug, Serialize, Default)]
+pub struct ApiUpdateMonitorCheckIn {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub status: Option<ApiMonitorCheckInStatus>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub duration: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub environment: Option<String>,
+}
```

Comment:
Renamed all of these to `Api*` since there's now some naming conflation with the structs provided in the `sentry-types` crate.